### PR TITLE
test(core): Add benchpress test for transplanted view change detection

### PR DESCRIPTION
**Pull request by atscott**
Adds a performance benchmark for transplanted views views (those which are declared in one component and inserted into another).

### CODE REVIEW

**Review by gkalpak — APPROVED** [review]
Commit message typo: `views views` --> `views` plus a couple of minor questions/suggestions.
Otherwise lgtm ✨

File: modules/benchmarks/src/change_detection/transplanted_views/transplanted_views.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+import {ChangeDetectionStrategy, Component, Input, NgModule, TemplateRef, ɵdetectChanges} from '@angular/core';
+import {numViews} from '../util';
+
+@Component({
+  selector: 'declaration-component',
+  template: `
+  <ng-template #template>{{trackTemplateRefresh()}}</ng-template>
+  <ng-container *ngFor="let n of views; trackBy: trackByIndex">
+    <insertion-component [template]="template" [viewCount]="viewCount"></insertion-component>
+  </ng-container>
```
Comment (mhevery) [inline modules/benchmarks/src/change_detection/transplanted_views/transplanted_views.ts]:
```suggestion
    <insertion-component *ngFor="let n of views; trackBy: trackByIndex" [template]="template" [viewCount]="viewCount"></insertion-component>
```

Comment (mhevery) [reply modules/benchmarks/src/change_detection/transplanted_views/transplanted_views.ts]: perhaps remove it here and just use the `*ngFor` inside the `InsertionComponent`

File: modules/benchmarks/src/change_detection/transplanted_views/transplanted_views.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  `,
+})
+export class DeclarationComponent {
+  views: undefined[] = [];
+  @Input()
+  set viewCount(n: number) { this.views = new Array(n); }
+  get viewCount() { return this.views.length; }
```
Comment (mhevery) [inline modules/benchmarks/src/change_detection/transplanted_views/transplanted_views.ts]: clean up the component as there are no longer many views.

File: modules/benchmarks/src/change_detection/transplanted_views/transplanted_views.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  template: `
+    <ng-container *ngFor="let n of views; template: template; trackBy: trackByIndex"></ng-container>
+  `,
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class InsertionComponent {
+  @Input() template !: TemplateRef<{}>;
+  views: undefined[] = [];
+  @Input()
+  set viewCount(n: number) { this.views = new Array(n); }
```
Comment (mhevery) [inline modules/benchmarks/src/change_detection/transplanted_views/transplanted_views.ts]: `new Array(x)` creates a holey array which has slower access than proper array. In this case it probably makes no difference however, I am worried about showing the right example. Replace with `newArray` helper.

File: modules/benchmarks/src/change_detection/transplanted_views/transplanted_views.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+  trackByIndex(index: number, item: any) { return index; }
```
Comment (mhevery) [inline modules/benchmarks/src/change_detection/transplanted_views/transplanted_views.ts]: leave a comment as to why this is here.

**Review by mhevery — APPROVED** [review]
Clean up

**Pull request closed without merging**